solr/core/src/java/org/apache/solr/cluster/events/impl/DelegatingClusterEventProducer.java

```diff
@@ -144,7 +152,25 @@ public synchronized void stop() {
   }
 
   @VisibleForTesting
-  public int getVersion() {
-    return version;
+  public int waitForVersionChange(int currentVersion, int timeoutSec) throws InterruptedException, TimeoutException {
```

Comment:
I debated whether to add this to the wrappers... it's only needed in tests. OTOH putting it here makes the test code much simpler.